apps/docs/content/guides/auth/social-login/auth-google.mdx

```diff
@@ -341,6 +341,83 @@ Most web apps and websites can utilize Google's [personalized sign-in buttons](h
    // Use 'nonce' when invoking the supabase.auth.signInWithIdToken() method
    ```
 
+### One-tap with NextJS
+
+If you're integrating Google One-Tap with your NextJS application, you can refer to the example below to get started:
+
+```tsx
+'use client'
+
+import Script from 'next/script'
+import { createClient } from '@/utils/supabase/client'
+import { CredentialResponse } from 'google-one-tap'
+import { useRouter } from 'next/navigation'
+import { useEffect } from 'react'
+
+const OneTapComponent = () => {
+  const router = useRouter()
+
+  // generate nonce to use for google id token sign-in
+  const generateNonce = async (): Promise<string[]> => {
+    const nonce = btoa(String.fromCharCode(...crypto.getRandomValues(new Uint8Array(32))))
+    const encoder = new TextEncoder()
+    const encodedNonce = encoder.encode(nonce)
+    const hashBuffer = await crypto.subtle.digest('SHA-256', encodedNonce)
+    const hashArray = Array.from(new Uint8Array(hashBuffer))
+    const hashedNonce = hashArray.map((b) => b.toString(16).padStart(2, '0')).join('')
+
+    return [nonce, hashedNonce]
+  }
+
+  useEffect(() => {
+    const initializeGoogleOneTap = () => {
+      window.onload = async () => {
```

Comment:
Hmm why is this `window.onload` necessary? Prefer using `addEventListener` in `useEffect` so you can remove it as React in strict mode runs the effect twice. 😭